deferred.go

```diff
@@ -24,10 +31,12 @@ func (d *Deferred) MarshalCBOR(w io.Writer) error {
 }
 
 func (d *Deferred) UnmarshalCBOR(br io.Reader) (err error) {
-	// Reuse any existing buffers.
-	reusedBuf := d.Raw[:0]
-	d.Raw = nil
-	buf := bytes.NewBuffer(reusedBuf)
```

Comment:
Checking for understanding:

In the old code `reusedBuf` was overallocated compared to the data in `d.Raw`, i.e. capacity was greater than length.  When we assign `reusedBuf` to the slice of `Raw` the over capacity is carried along with it.  Then when we pass `reusedBuf` to `bytes.NewBuffer` it further overallocates more capacity than is needed by `reusedBuf`, amplifying the allocation further.  

Is this correct?

Reply:
Mostly the second part. In the case we care about, `reusedBuf` was always empty. But `bytes.Buffer` generally over-allocates for better performance.